rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -141,6 +169,38 @@ void Recorder::record()
   }
 }
 
+void Recorder::event_publisher_thread_main()
+{
+  RCLCPP_INFO(get_logger(), "Event publisher thread: Starting");
+
+  bool should_exit = false;
+
+  while (!should_exit) {
+    std::unique_lock<std::mutex> lock(event_publisher_thread_mutex_);
+    event_publisher_thread_wake_cv_.wait(
+      lock,
+      [this] {return event_publisher_thread_should_wake();});
+
+    if (write_split_has_occurred_) {
+      write_split_has_occurred_ = false;
+
+      auto message = rosbag2_interfaces::msg::WriteSplitEvent();
+      message.closed_file = bag_split_info_.closed_file;
+      message.opened_file = bag_split_info_.opened_file;
+      split_event_pub_->publish(message);
+    }
+
+    should_exit = event_publisher_thread_should_exit_;
```

Comment:
Perhaps I'm missing something, but where is `event_publisher_thread_should_exit_` being set/modified after line 112? Is this going to be always `true`?

Reply:
Yes. The thread exists from the start of recording for the remaining lifetime of the `Recorder` and only exits when the recorder destructs. This is consistent with the way the `Recorder` class, once it starts recording, only stops recording when it destructs.